### PR TITLE
PAINTROID-151: Redesign copy&paste tool options

### PR DESCRIPTION
**Pull request by StanglTobias**
*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by Electronix1337 — CHANGES_REQUESTED** [review]
I really like the new design of the stamp tool options! But some (hopefully) minor things have to be changed in my opinion. By the way the deletion of the long click is reasonable in my opinion. 

1) If we copy a box content and rotate the phone the paste button is disabled, even though the stamp tool box has the content before rotating.
2) At least for me on API <= 23 the new buttons are not displaying any text at all. [https://ibb.co/Z1Hc5nw](url) and I would also suggest to have a small space between the "Save" and "Discard" button.
3) I'm not quite sure where this happens, but the selected color in the color picker option is not displayed in the background of the "Apply" button. Background of "Apply" button always stays in the same color except when choosing "transparent" then the color is changed to white otherwise, only the text is changed from black to white depending on the selected color.
4) Is there a special reason why "Cut" option was not implemented? We could create an own ticket for that if it's too much for one ticket but I think the feature can be extremely helpful for users.
5) In my opinion it's quite unintuitive how the tool options overall behave, why not just close the tool options window if user presses back and they are currently opened? And we might consider disabling the onclick for stamp tool options, because the options have to be present all the time in order to actively use the tool, so I don't know if hiding the stamp tool options makes sense. It's definitely just a matter of taste but for me it's rather disturbing than helpful.

**Comment by StanglTobias** [comment]
>     4. Is there a special reason why "Cut" option was not implemented? We could create an own ticket for that if it's too much for >         one ticket but I think the feature can be extremely helpful for users.

There is a seperate ticket for the cut option.


>     5. In my opinion it's quite unintuitive how the tool options overall behave, why not just close the tool options window if user 
>         presses back and they are currently opened? And we might consider disabling the onclick for stamp tool options, because >         the options have to be present all the time in order to actively use the tool, so I don't know if hiding the stamp tool options 
>         makes sense. It's definitely just a matter of taste but for me it's rather disturbing than helpful.

I totally agree with this, but this has nothing in common with this PR. It should be treated in a separate ticket since this one PR did not change the behavior to this.

File: Paintroid/src/main/java/org/catrobat/paintroid/presenter/MainActivityPresenter.java [hunk 1]
```diff
@@ -431,8 +431,6 @@ public void onBackPressed() {
 			drawerLayoutViewHolder.closeDrawer(Gravity.END, true);
 		} else if (model.isFullscreen()) {
 			exitFullscreenClicked();
-		} else if (toolController.toolOptionsViewVisible()) {
-			toolController.hideToolOptionsView();
```
Comment (Electronix1337) [inline Paintroid/src/main/java/org/catrobat/paintroid/presenter/MainActivityPresenter.java:435]: Errors are all fixed, but back to 5) aren't that the lines of code for instantly setting the brush tool when pressing back? On develop branch the behavior is as desired: If option window is open and user presses back, the option window is closed. I would also prefer that brush tool option window gets closed when user touches the drawing surface but that's just a matter of taste again. Going to put it into PO review and let Wolfgang decide if changes are needed or not.

**Review by Electronix1337 — APPROVED** [review]

**Review by Electronix1337 — CHANGES_REQUESTED** [review]
Actually one little thing has to be fixed, for some reason the "transparent" color is displayed as white now. Maybe fix this before I drag the ticket into PO review.

Colorpicker on develop branch:

color_picker_normal

Colorpicker in this ticket: 

color_picker_transparent_wrong

**Comment by StanglTobias** [comment]
all tests pass and my local device.

**Review by Electronix1337 — APPROVED** [review]
lgtm, flaky tests on Jenkins, seem to depend on the slave/state of Jenkins, whether these tests fail or not.

**Comment by wslany** [comment]
Thanks!

**Pull request merged**